### PR TITLE
Thermalctld APIs for recording min and max temp

### PR DESCRIPTION
**Pull request by mprabhu-nokia**
sonic-platform-base: Changes to introduce APIs for modular chassis for thermalctld

HLD: Azure/SONiC#646

Introducing thermal APIs to get min and max temperatures of each sensors
  - get_minimum_recorded()
  - get_maximum_recorded()

### CODE REVIEW

File: sonic_platform_base/thermal_base.py [hunk 1]
```diff
@@ -92,3 +92,23 @@ def get_low_critical_threshold(self):
...
+
+    def get_minimum_recorded(self):
+        """
+        Retrieves the minimum recorded temperature of thermal
+
+        Returns:
+            A float number, the minimum recorded temperature of thermal in Celsius
+            up to nearest thousandth of one degree Celsius, e.g. 30.125
+        """
+        raise NotImplementedError
```
Comment (mprabhu-nokia) [inline sonic_platform_base/thermal_base.py:104]: Have logic in thermalctld to calculate this for backward compatibility.

Comment (mprabhu-nokia) [reply sonic_platform_base/thermal_base.py:104]: Using try_get() to keep it optional.

**Comment by anshuv-mfst** [comment]
@judyjoseph - could you please review.

**Comment by judyjoseph** [comment]
LGTM

**Review by judyjoseph — APPROVED** [review]

**Comment by jleveque** [comment]
Please fix conflicts.

**Comment by mprabhu-nokia** [comment]
@jleveque Could you please review this as well?

**Review by jleveque — APPROVED** [review]

**Comment by jleveque** [comment]
@mprabhu-nokia: Please also add corresponding tests for these APIs to the platform API tests in the sonic-mgmt repo.

**Comment by mprabhu-nokia** [comment]
> 
> 
> @mprabhu-nokia: Please also add corresponding tests for these APIs to the platform API tests in the sonic-mgmt repo.

Will do. Appreciate quick help with the reviews and merge.

**Pull request merged**